readthedocs/core/utils/url.py

```diff
@@ -7,7 +7,13 @@ def join_url_path(base, *args):
 
     This does a simple join of the base path with the path components,
     inserting a slash between each component.
-    It does not offer protection against directory traversal attacks.
+    The resulting path will always start with a slash.
+
+    .. warning::
+
+       This does not offer protection against directory traversal attacks,
+       it simply joins the path components together. This shouldn't be used
+       to serve files, use ``readthedocs.storage.utils.safe_join`` for that.
```

Comment:
💯

Reply:
Should we name this function `unsafe_ join_url_path`, so it's obvious?